pkg/client/events_query/connection.go

```diff
@@ -0,0 +1,34 @@
+package eventsquery
+
+import (
+	"github.com/gorilla/websocket"
+
+	"pocket/pkg/client"
+)
+
+var _ client.Connection = &websocketConn{}
+
+// websocketConn implements the Connection interface using the gorilla websocket
+// transport implementation.
+type websocketConn struct {
+	conn *websocket.Conn
+}
+
+// Receive implements the respective interface method using the underlying websocket.
+func (wsConn *websocketConn) Receive() ([]byte, error) {
+	_, msg, err := wsConn.conn.ReadMessage()
```

Comment:
Shouldn't we guard this method with a `Lock`?

Reply:
To protect what? The connection is assigned by the dialer. If the connection is closed it should return an error but it's location in memory isn't changing.

Reply:
Was imagining concurrent calls to `Receive`... not sure how `ReadMessage` would react in that case.

Reply:
🤔 We shouldn't be calling `ReadMessage()` concurrently on the same connection anywhere, each connection is read in its own for loop. I would assume that concurrent calls to read would result in one caller getting a return and the rest blocking. I could be mistaken, I'll take a look at the code when I get a chance. Regardless, I think it's unlikely that there's a data race because I would expect `ReadMessage` to return a new value or reference each time. Again, I could be mistaken; will have to look at the code